src/Crdt/SyncHelper.cs

```diff
@@ -0,0 +1,77 @@
+using System.Text.Json;
+using Crdt.Core;
+
+namespace Crdt;
+
+public static class SyncHelper
+{
+    /// <summary>
+    /// simple sync example, each ISyncable could be over the wire or in memory
+    /// prefer that remote is over the wire for the best performance, however they could both be remote
+    /// </summary>
+    /// <param name="localModel"></param>
+    /// <param name="remoteModel"></param>
+    /// <param name="serializerOptions"></param>
+    internal static async Task<SyncResults> SyncWith(ISyncable localModel,
+        ISyncable remoteModel,
+        JsonSerializerOptions serializerOptions)
+    {
+        if (!await localModel.ShouldSync() || !await remoteModel.ShouldSync()) return new SyncResults([], [], false);
+        var localSyncState = await localModel.GetSyncState();
+
+        var (missingFromLocal, remoteSyncState) = await remoteModel.GetChanges(localSyncState);
+        //todo abort if local and remote heads are the same
+        var (missingFromRemote, _) = await localModel.GetChanges(remoteSyncState);
+        if (localModel is DataModel && remoteModel is DataModel)
+        {
+            //cloning just to simulate the objects going over the wire
```

Comment:
When and how can we remove this in situations when it actually does go over the wire?

Reply:
it already is. When it's going over the wire both the local and remote models won't be an instance of DataModel, In FLEx lite the remote model is implemented with the `CrdtProjectSync` class (which implements the ISyncable interface), that class syncs over http so the data has to be serialized.

This case is pretty much only relevant for unit tests where both remote and local are in memory. It might also be relevant for syncing to a flashdrive depending on how that's implemented.